src/v/storage/log_manager.cc

```diff
@@ -196,8 +196,16 @@ ss::future<>
 log_manager::housekeeping_scan(model::timestamp collection_threshold) {
     using bflags = log_housekeeping_meta::bitflags;
 
-    if (_logs_list.empty()) {
-        co_return;
+    static constexpr auto is_not_set = [](bflags var, auto flag) {
+        return (var & flag) != flag;
+    };
```

Comment:
the previous "not set" was comparing against `none`

`(_logs_list.front().flags & bflags::compacted) == bflags::none`

was this because there were only 2 flags? maybe the assertion failure you saw in CI is related?

Reply:
yes only 0 and 1. don't think it was related, also because the code before this was correct, it would reset flags back to none so no risk of uninitialized variables spoiling the check